prov/efa/src/efa_fabric.c

```diff
@@ -479,13 +489,30 @@ static int efa_get_gdr_support(char *device_name)
 		goto out;
 	}
 
-	ret = fi_read_file(get_sysfs_path(), gdr_path, &gdr_support, sizeof(char));
-	if (ret < 0)
+	memset(gdr_support_str, 0, sizeof(gdr_support_str));
```

Comment:
why memset here?

Reply:
To make sure gdr_support_str is terminated with '\0'. My understanding is ```fi_read_file``` will not append '\0' to the end of string.

Reply:
This is a very expensive way to do that.  And not very useful.  What if two characters are read from the file?  I think what you want to do is:

```C
  ret = fi_read_file(..., gdr_support_str, MAX_GDR_SUPPORT_STRLEN);
  if (ret < 0) {
    /* error handling */
  }
  /* fi_read_file does not guarantee NULL terminated string.  Guarantee it. */
  gdr_support_str[MAX_GDR_SUPPORT_STRLEN - 1] = '\0';
```
This still isn't great, as you don't know if you read (say) a 12 instead of a 1.  But it is significantly better than what you have.

Reply:
I see, thanks! How about

```
static const int MAX_GDR_SUPPORT_STRLEN = 80;. 
ret = fi_read_file(..., gdr_support_str, MAX_GDR_SUPPORT_STRLEN-1);
if (ret <0) {
   /* error handling */
}
gdr_support_str[ret] = '\0';
```
This would handle the case we read say 12 back.